packages/uilib/src/lib/plugins/communication-explorer/telemetry-view/telemetry-view.svelte

```diff
@@ -1,73 +1,85 @@
-	export let showSidebar = true
-	
-	let rootNode: RootNode | undefined = undefined
-	$: initInfos(root, $filterState, $preferences$)
-	let lastUsedRoot: Element | undefined = undefined
-	let lastExtractedInfos: IEDCommInfo[] = []
-	
-	// Note: maybe have a mutex if there are too many changes
-	async function initInfos(
-		root: Element,
-		selectedFilter: SelectedFilter,
-		preferences: Preferences,
-	) {
-		if (!root) {
-			console.info({ level: "info", msg: "initInfos: no root" })
-			return []
-		}
-		
-		if(root !== lastUsedRoot) {
-			const iedInfos = extractIEDInfos(root)
-			lastExtractedInfos = iedInfos
-			lastUsedRoot = root
-		}
-		rootNode = await calculateLayout(lastExtractedInfos, config, selectedFilter, preferences)
-	}
-	
-	const config: Config = {
-		width:  150,
-		height: 40,
-		// spacingBetweenNodes: 100,
-		// spacingBase: 40,
-		// heightPerConnection: 20,
-	}
-	
-	function handleIEDSelect(e: CustomEvent<IEDNode>) {
-		selectIEDNode(e.detail)
-	}
-	function handleIEDAdditiveSelect(e: CustomEvent<IEDNode>) {
-		toggleMultiSelectionOfIED(e.detail)
-	}
-	function handleConnectionClick(e: CustomEvent<IEDConnection>) {
-		// temp till fully migrated: map element to enhanced data model
-		const selectedConnection = e.detail as IEDConnectionWithCustomValues
-		selectConnection(selectedConnection)
+import { calculateLayout } from '../_func-layout-calculation/node-layout'
+import {
+	Diagram,
+	type IEDConnection,
+	type IEDConnectionWithCustomValues,
+	type IEDNode,
+	type RootNode
+} from '../../../components/diagram'
+import { Sidebar } from '../sidebar'
+import {
+	filterState,
+	type SelectedFilter,
+	selectConnection,
+	selectIEDNode,
+	clearIEDSelection,
+	toggleMultiSelectionOfIED
+} from '../_store-view-filter'
+import type { Config } from '../_func-layout-calculation/config'
+import { preferences$, type Preferences } from '../_store-preferences'
+// SERVICES
+import { getIEDCommunicationInfos } from '../services/ied'
+// TYPES
+import type { IED } from '@oscd-plugins/core'
+
+//
+// INPUT
+//
+export let root: Element
+export let showSidebar = true
+
+let rootNode: RootNode | undefined = undefined
+$: initInfos(root, $filterState, $preferences$)
+let lastUsedRoot: Element | undefined = undefined
+let lastExtractedInfos: IED.CommunicationInfo[] = []
+
+// Note: maybe have a mutex if there are too many changes
+async function initInfos(
+	root: Element,
+	selectedFilter: SelectedFilter,
+	preferences: Preferences
+) {
+	if (!root) {
+		console.info({ level: 'info', msg: 'initInfos: no root' })
+		return []
 	}
-	function handleClearClick() {
-		clearIEDSelection()
+
+	if (root !== lastUsedRoot) {
+		const iedInfos = getIEDCommunicationInfos(root)
+		lastExtractedInfos = iedInfos
+		lastUsedRoot = root
 	}
-	
+	rootNode = await calculateLayout(
+		lastExtractedInfos,
+		config,
+		selectedFilter,
+		preferences
+	)
+}
+
+const config: Config = {
+	width: 150,
+	height: 40
+	// spacingBetweenNodes: 100,
+	// spacingBase: 40,
```

Comment:
can be removed if not needed oder?

Reply:
I think you are right, but this plugin is out of my scope: I have updated the plugin with the new core features, and I would rather avoid making any other changes for now.
So I will not update it and I will leave it to @interBN or @pwinkelholzS1 if they are working on this side of the plugin.